Bump rustfmt to nightly version and run it.

jaccarmac opened the pull request:
The `rustfmt` crate is no longer updated however, it is only kept around for compatibility with stable Rust. See https://users.rust-lang.org/t/rustfmt-releases/11357 and the linked blog post: New development is happening on a new crate which is recommended.

Of course, this requires a one-time reformat of all the Remacs code and an update to developers' configurations, which might outweigh the benefit of getting the new lints.

If one follows the `rustfmt` team's recommendation and uses the `rustfmt-nightly` crate it creates a lot of inconvenient diffs when developing. I believe the one-time pain is worth being on the latest and maintained formatter.

jaccarmac commented:
Missed the fact that the project is using the older version of `rustfmt`, my bad. Are there any plans to update to the `rustfmt-nightly` version as Remacs already uses nightly Rust?

shanavas786 commented on .travis.yml:
```diff
@@ -35,7 +35,7 @@ jobs:
       # Install rustfmt 0.9.0 if it isn't already installed on
       # Travis. This can be slow to install, so raise the Travis timeout.
-      before_script: (which rustfmt && rustfmt --version && [[ "$(rustfmt --version)" =~ "0.9.0" ]]) || travis_wait cargo install --force rustfmt --vers 0.9.0
+      before_script: (which rustfmt && rustfmt --version && [[ "$(rustfmt --version)" =~ "0.2.9" ]]) || travis_wait cargo install --force rustfmt-nightly --vers 0.2.9
```
0.9.0 is the latest version.

jaccarmac replied:
The `rustfmt` crate is no longer updated however, it is only kept around for compatibility with stable Rust. See https://users.rust-lang.org/t/rustfmt-releases/11357 and the linked blog post: New development is happening on a new crate which is recommended.

Of course, this requires a one-time reformat of all the Remacs code and an update to developers' configurations, which might outweigh the benefit of getting the new lints.

shanavas786 replied:
Oh, I didn't notice `-nightly` in name :)

jaccarmac replied:
Updated the PR to reflect the actual intent, since I misunderstood what I was doing the first time I submitted it.

brotzeit replied:
I've also considered making a PR with the latest rustfmt version, but I thought there's a reason for using the stable version. I think it's a good idea :thumbsup:

birkenfeld commented:
New style is an improvement I think. So I'm +1.

shaleh commented:
Bite the bullet now. I just checked and I have 0.9.0-nightly and that is how all of my PRs have been formatted.

BTW, this may not be news to everyone but it was to me....

    git remote add request-author git://github.com/request-author/project.git
    git fetch request-author
    git checkout -b request-author/branch

Now you can check out that branch and test things nicely.

jaccarmac commented:
@shaleh: Just to be perfectly clear, `0.9.0-nightly` is *not* the "nightly" version in question and is older than the `0.2.9-nightly` in question. The 0.9.0 release is from the old `rustfmt` crate while the 0.2.9 version is from an entirely different crate, `rustfmt-nightly`. The version numbering doesn't really help anyone.

shaleh commented:
!OH!

Thanks.

On October 24, 2017 8:07:13 PM PDT, Jacob MacDonald <notifications@github.com> wrote:
>@shaleh: Just to be perfectly clear, `0.9.0-nightly` is *not* the
>"nightly" version in question and is older than the `0.2.9-nightly` in
>question. The 0.9.0 release is from the old `rustfmt` crate while the
>0.2.9 version is from an entirely different crate, `rustfmt-nightly`.
>The version numbering doesn't really help anyone.
>
>-- 
>You are receiving this because you were mentioned.
>Reply to this email directly or view it on GitHub:
>https://github.com/Wilfred/remacs/pull/394#issuecomment-339200256

-- 
Sent from my Android device with K-9 Mail. Please excuse my brevity.

jaccarmac commented:
Updated the code pushed since I opened this and updated the documentation which I had neglected before. Merge conflicts are going to be a thing as long as the PR is around unfortunately, such is the nature of cosmetic tweaking :/

shaleh commented:
So if one of us clicks "merge" does Travis automatically update? Or does this break everyone?

jaccarmac commented:
Travis will automatically update and I believe existing successful Travis runs will still show as successful. However, all future builds will be unsuccessful unless formatted in the new style and many pull requests will gain merge conflicts.

shaleh commented:
I am eager to either put this to bed or ditch it. No one can sanely keep up.

birkenfeld commented:
Well, once conflicts are resolved clicking "merge" is easy :)

shaleh commented:
I almost clicked go on this. But I did not want to break Travis. I have zero experience with Travis.

jaccarmac commented:
Shall I rebase once again? I'd prefer to do it only once as basically any change is going to force me to redo it.

jaccarmac commented:
@shaleh: This does not break Travis. In fact, since I updated `travis.yml` this PR ran with the new configuration! Check out the check at https://travis-ci.org/Wilfred/remacs/builds/292908944?utm_source=github_status&utm_medium=notification. That was the commit before the rebase I'm about to do. As you can see, it's a normal Travis build for Remacs with a new version of `rustfmt-nightly` and some updated code to make that `rustfmt` check pass.

birkenfeld commented:
OK, let's see...

jaccarmac commented:
Whoops, just rebased and tried to push. Assuming the merge succeeded I'll back off and let yours go through :)

jaccarmac commented:
Just realized the build is failing with the same error @shaleh ran into earlier which forces cargo to install `rustfmt` each time. Pushing a fix, my bad not catching that.

jaccarmac commented:
Hopefully that didn't bust the build...

shaleh commented:
We need a note in the README about how to have a working rustfmt.

On October 25, 2017 10:19:19 PM PDT, Jacob MacDonald <notifications@github.com> wrote:
>Just realized the build is failing with the same error @shaleh ran into
>earlier which forces cargo to install `rustfmt` each time. Pushing a
>fix, my bad not catching that.
>
>-- 
>You are receiving this because you were mentioned.
>Reply to this email directly or view it on GitHub:
>https://github.com/Wilfred/remacs/pull/394#issuecomment-339554223

-- 
Sent from my Android device with K-9 Mail. Please excuse my brevity.

jaccarmac commented:
Should be added now. But hang on, I don't understand Travis as well as I thought I did.

jaccarmac commented:
Formatting checks pass. My latest commit is a duplicate of #416. Gonna leave this for tonight though, if it needs cleaning tomorrow I will do it then.
